backend/src/main/scala/cromwell/backend/BackendWorkflowInitializationActor.scala

```diff
@@ -2,58 +2,127 @@ package cromwell.backend
   }
 
   /**
     * Our predefined sequence to run during preStart
     */
-  final def initSequence = for {
-    _ <- validate
-    _ <- beforeAll
+  final def initSequence() = for {
+    _ <- validate()
+    _ <- beforeAll()
   } yield InitializationSuccess
 
   /**
     * Abort all initializations.
     */
-  def abortInitialization: Future[WorkflowAbortResponse]
+  def abortInitialization(): Future[WorkflowAbortResponse]
+
+  /**
+    * A call which happens before anything else runs
+    */
+  def beforeAll(): Future[Unit]
+
+  /**
+    * Validates runtime attributes for one specific call.
+    *
+    * @param runtimeAttributes Runtime Attributes with already evaluated values.
+    * @return If all entries from runtime attributes section are valid Success otherwise
+    *         Failure with the aggregation of errors.
+    */
+  def validateRuntimeAttributes(runtimeAttributes: EvaluatedRuntimeAttributes): Future[scalaz.Validation[NonEmptyList[String], Unit]]
 
   /**
     * Validate that this WorkflowBackendActor can run all of the calls that it's been assigned
     */
-  def validate: Future[Unit]
+  protected def validate(): Future[Unit] = {
+    def eval(call: Call) = validateRuntimeAttributes(evaluateRuntimeAttributesWdlExpressions(call))
+
+    val validationResult: Future[Seq[(Call, Validation[NonEmptyList[String], Unit])]] =
+      Future.sequence(calls map { call => (call, eval(call)) } map { entry => entry._2.map(valRes => (entry._1, valRes)) })
+
+    val failedValidations = validationResult.map(evaluateValidationResultPerCall)
+
+    //TODO: Add recover or not but refactor this as part of https://github.com/broadinstitute/cromwell/issues/725.
+    failedValidations map { failures =>
+      if (failures.nonEmpty) {
+        throw new AggregatedException("Runtime attribute validation failed", failures)
+      }
+    }
+  }
 
   /**
-    * A call which happens before anything else runs
+    * Creates WDL expression evaluator for the specific call.
+    *
+    * @param call Call which contains WDL expressions in it.
+    * @return A WDL ValueEvaluator.
+    */
+  protected def createEvaluator(call: Call): ValueEvaluator = {
+    val declarations = workflowDescriptor.workflowNamespace.workflow.declarations ++ call.task.declarations
+    val knownInputs = workflowDescriptor.inputs
+    val lookup = WdlExpression.standardLookupFunction(knownInputs, declarations, NoFunctions)
+    new ValueEvaluator(lookup, NoFunctions)
+  }
+
+  /**
+    * Evaluates WDL expressions within call runtime attributes.
+    *
+    * @param call Call which contains WDL expressions in it.
+    * @return Evaluated runtime attributes.
     */
-  def beforeAll: Future[Unit]
+  private def evaluateRuntimeAttributesWdlExpressions(call: Call): EvaluatedRuntimeAttributes = {
+    val evaluator = createEvaluator(call)
+    def evaluate(wdlExpression: WdlExpression) = evaluator.evaluate(wdlExpression.ast)
```

Comment:
I personally think we should prefer to not use `ValueEvaluator`s directly because it exposes ASTs unnecessarily.  Instead maybe `createEvaluator(call: Call): ValueEvaluator` could become `lookupFunction(call: Call)(variable: String): WdlValue`

Then this line could become `wdlExpression.evaluate(lookupFunction(call), NoFunctions)`